core/io/net_socket.h

```diff
@@ -34,6 +34,16 @@
 #include "core/io/ip.h"
 #include "core/reference.h"
 
+typedef struct {
+#ifdef WINDOWS_ENABLED
+	size_t dataLength;
+	void *data;
+#else
+	void *data;
+	size_t dataLength;
+#endif
+} NetSocketBuffer;
+
```

Comment:
On windows, the properties in this struct are inverted. (WSABUF https://docs.microsoft.com/en-us/windows/win32/api/ws2def/ns-ws2def-wsabuf)